src/components/BarChart/BarChart.jsx

```diff
@@ -0,0 +1,297 @@
+		/**
+		 * There are some cases where you need to only show a "sampling" of ticks
+		 * on the x axis. This number will control that.
+		 */
+		xAxisTickCount: number,
+		/**
+		 * An optional function used to format your x axis data. If you don't
+		 * provide anything, we'll use an identity function.
+		 */
+		xAxisFormatter: func,
+		/**
+		 * Set a title for the x axis.
+		 */
+		xAxisTitle: string,
+		/**
+		 * Set a color for the x axis title. This takes any number 0 or greater and
+		 * it converts it to a color in our color palette.
+		 */
+		xAxisTitleColor: number,
+
+
+		/**
+		 * An array of your y axis fields. Typically this will just be a single
+		 * item unless you need to display grouped or stacked bars.
+		 */
+		yAxisFields: array.isRequired,
+		/**
+		 * The minimum number the y axis should display. Typically this
+		 * should be be `0`.
+		 */
+		yAxisMin: number,
+		/**
+		 * The maximum number the y axis should display. This should almost always
+		 * be the largest number from your dataset.
+		 */
+		yAxisMax: number,
+		/**
+		 * An optional function used to format your y axis data. If you don't
+		 * provide anything, we use the default D3 number formatter.
+		 */
+		yAxisFormatter: func,
+		/**
+		 * Stack the y axis data instead of showing it as groups. This is only
+		 * useful if you have multiple `yAxisFields`. Stacking will cause the chart
+		 * to be aggregated by sum.
+		 */
+		yAxisIsStacked: bool,
+		/**
+		 * There are some cases where you need to only show a "sampling" of ticks
+		 * on the y axis. This number will control that.
+		 */
+		yAxisTickCount: number,
+		/**
+		 * Set a title for the y axis.
+		 */
+		yAxisTitle: string,
+		/**
+		 * Set a color for the y axis title. This takes any number 0 or greater and
+		 * it converts it to a color in our color palette.
+		 */
+		yAxisTitleColor: number,
+	},
+
+	getDefaultProps() {
+		return {
+			height: 400,
+			width: 1000,
+			margin: {
+				top: 10,
+				right: 20,
+				bottom: 50,
+				left: 80,
+			},
+
+			xAxisField: 'x',
+			xAxisTickCount: null,
+			xAxisTitle: null,
+			xAxisTitleColor: -1,
+
+			yAxisFields: ['y'],
+			yAxisTickCount: null,
+			yAxisIsStacked: false,
+			yAxisMin: 0,
+			yAxisTitle: null,
+			yAxisTitleColor: -1,
+		};
+	},
+
+	render() {
+		const {
+			className,
+			height,
+			width,
+			margin,
+			data,
+
+			xAxisField,
+			xAxisFormatter,
+			xAxisTitle,
+			xAxisTitleColor,
+			xAxisTickCount,
+
+			yAxisFields,
+			yAxisFormatter,
+			yAxisTitle,
+			yAxisTitleColor,
+			yAxisIsStacked,
+			yAxisTickCount,
+			yAxisMin,
+			yAxisMax = yAxisIsStacked
+				? maxByFieldsStacked(data, yAxisFields)
+				: maxByFields(data, yAxisFields),
+
+			...passThroughs,
+		} = this.props;
+
+		if (_.isEmpty(data)) {
+			return null;
+		}
+
+		const innerWidth = width - margin.left - margin.right;
+		const innerHeight = height - margin.top - margin.bottom;
+
+		const paddingInner = yAxisFields.length > 1 ? 0.3 : 0.05;
```

Comment:
I'll add constants to make it more clear what's going on. These numbers control the spacing between bars.